src/main/java/com/leverx/learningmanagementsystem/config/security/SecurityConfiguration.java

```diff
@@ -0,0 +1,90 @@
+package com.leverx.learningmanagementsystem.config.security;
+
+import com.sap.cloud.security.xsuaa.XsuaaServiceConfiguration;
+import com.sap.cloud.security.xsuaa.token.TokenAuthenticationConverter;
+import lombok.AllArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.annotation.Order;
+import org.springframework.security.config.Customizer;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+import org.springframework.security.web.SecurityFilterChain;
+
+@Configuration
+@EnableWebSecurity
+@AllArgsConstructor
+public class SecurityConfiguration {
```

Comment:
There are 2 approaches:
1. Define all SecurityFilterChains in one config class
2. Create separate class for each SecurityFilterChain

The second approach has benefit when configuration becomes complicated or we have to define big amount of beans related to target SecurityFilterChain. Otherwise we have mix of beans inside of one config class that are related to different SecurityFilterChains

Also it's better to split local and cloud configs by classes